documentation/guides/migrating-from-v3-to-v4.md

```diff
@@ -72,9 +72,64 @@ Polaris now supports multiple languages and ships with [many translations](https
 <AppProvider />
 
 // new
+<AppProvider i18n={translations}>
+```
+
+`translations` can be handled in different ways, depending on your needs.
+
+#### Statically
+
+You can specify translation file explicitly:
+
+```jsx
 import translations from '@shopify/polaris/locales/en.json';
+```
 
-<AppProvider i18n={translations}>
+This approach is suitable:
+
+- when project have only one specific locale
+- inside tests when you don't care which translations to use and need to pass any
+
+#### Dynamically
+
+What if your project supports different locales? Don't import all translation files as in example above - it's not optimal, you will end up with all translations in your bundle. What you need is to dynamically choose translations. There are 2 options.
+
+First option: you can copy/paste [translation files](https://github.com/Shopify/polaris-react/tree/master/locales) from Polaris into your project , into `translations` folder in the same place where your `AppProvider` is used. Then internationalization hook will work as usual and automatically will pick them up. You have freedom of controlling all translations, however, manual copy/pasting and updating is required.
```

Comment:
I really dislike this first option - it means consumers have to constantly check for new translations  that are added to polaris, which feels like a complicated waste of time.

Can you remove this, and push people towards the second option?